crates/server/src/authority/catalog.rs

```diff
@@ -502,15 +536,15 @@ async fn build_response(
             .await
         }
         ZoneType::Forward | ZoneType::Hint => {
-            send_forwarded_response(future, request_header, &mut response_header).await
+            send_forwarded_response(result, request_header, &mut response_header).await
         }
     };
 
-    (response_header, sections)
+    LookupResult::Ok((response_header, sections))
 }
 
 async fn send_authoritative_response(
-    future: impl Future<Output = Result<Box<dyn LookupObject>, LookupError>>,
+    response: LookupResult<Box<dyn LookupObject>>,
```

Comment:
The change to store a `Result` directly here instead of a future seems unrelated and should ideally be at least in a separate commit (which only works if you squash all the tri-variant enum-related commits into a single one).

Reply:
This is related to being able to handle the Skip case [here](https://github.com/hickory-dns/hickory-dns/pull/2160/files/4ceeaca0d18fb64b233c4e71bd8ec339bb66764d#diff-b53c96460a7227a433d7322365b838c2383c65d61a088a0d41e31c292edc9940R517)